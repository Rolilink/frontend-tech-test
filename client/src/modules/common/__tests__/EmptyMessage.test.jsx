import React from 'react';
import { shallow } from 'enzyme';
import serializer from 'enzyme-to-json/serializer';

import EmptyMessage from '../EmptyMessage';

const renderMessage = () => <div className="message" />;

expect.addSnapshotSerializer(serializer);

describe('Common.EmptyMessage', () => {
  it('should render children when isEmpty is false', () => {
    const isEmpty = false;
    const children = <div className="children" />;

    const emptyMessage = shallow(
      <EmptyMessage isEmpty={isEmpty} renderMessage={renderMessage}>
        {children}
      </EmptyMessage>,
    );

    expect(emptyMessage).toMatchSnapshot();
  });

  it('should render a message with renderMessage when isEmpty is true', () => {
    const isEmpty = true;
    const children = <div className="children" />;

    const emptyMessage = shallow(
      <EmptyMessage isEmpty={isEmpty} renderMessage={renderMessage}>
        {children}
      </EmptyMessage>,
    );

    expect(emptyMessage).toMatchSnapshot();
  });
});
