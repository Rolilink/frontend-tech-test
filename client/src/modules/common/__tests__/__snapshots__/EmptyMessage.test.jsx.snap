// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common.EmptyMessage should render a message with renderMessage when isEmpty is true 1`] = `
<div
  className="EmptyMessage "
>
  <div
    className="EmptyMessage-Sign"
  >
    Here goes an empty sign
  </div>
  <div
    className="EmptyMessage-Message"
  >
    <div
      className="message"
    />
  </div>
</div>
`;

exports[`Common.EmptyMessage should render children when isEmpty is false 1`] = `
<div
  className="children"
/>
`;
